examples/performance/include/traccc/efficiency/seeding_performance_writer.hpp

```diff
@@ -70,6 +61,16 @@ class seeding_performance_writer {
         }
     }
 
+    void init() {
```

Comment:
You added a bunch of `init()` calls in the clients, where they had none before.

Would it not be possible to make this `init()` function protected/private and make the object call it whenever necessary. (When the user calls a function that requires the object to be initialised, while it's not initialised yet.) Then the clients would only need to put their `add_cache(...)` calls behind `if` statements.

Reply:
Ok. I changed to call `init()` in `add_cache(...)` if the file is not initiailized.